feat(cloudwatch): Add L2 for Insight Rule + add contributor insights widget

seanyles-amzn opened the pull request:
Adds an L2 Insight Rule and a corresponding Insight Widget.  My team had a use for it so I wrote one up and thought I'd PR it.

This PR continues PRs [19534](https://github.com/aws/aws-cdk/pull/19534) and [13008](https://github.com/aws/aws-cdk/pull/13008).

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: 28d676fb0645a49a0cff5398a298eb54fd945c0a
* Result: FAILED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=5f02d106-0adf-4972-ae09-f55b51a254ed%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

TheRealAmazonKendra reviewed (dismissed):
New L2 constructs require an approved [RFC](https://github.com/aws/aws-cdk-rfcs) before a pull request will be reviewed. See our [state diagram for the RFC Process](https://github.com/aws/aws-cdk-rfcs#state-diagram). I've converted this into draft mode to indicate that it is not yet ready for review.

TheRealAmazonKendra commented:
> > New L2 constructs require an approved [RFC](https://github.com/aws/aws-cdk-rfcs) before a pull request will be reviewed. See our [state diagram for the RFC Process](https://github.com/aws/aws-cdk-rfcs#state-diagram). I've converted this into draft mode to indicate that it is not yet ready for review.
> 
> Shouldn't there already be approved RFCs if the previous two PRs made it to review?

I'm not sure when those were created but we've recently added this requirement.

kaizencc commented:
@seanyles-amzn, I chatted with @TheRealAmazonKendra offline and we can actually use the prior PRs as a design doc for this L2, thereby satisfying the requirement for an RFC before implementation. Before I review, however, do you still have the bandwidth to iterate on this PR?

The first thing to do is to fix the failing build. Once that is done, please ping me and I will help review this PR!

seanyles-amzn commented:
> @seanyles-amzn, I chatted with @TheRealAmazonKendra offline and we can actually use the prior PRs as a design doc for this L2, thereby satisfying the requirement for an RFC before implementation. Before I review, however, do you still have the bandwidth to iterate on this PR?
> 
> The first thing to do is to fix the failing build. Once that is done, please ping me and I will help review this PR!

Yes I still have the bandwidth to work this PR. I will update the build as soon as I can

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: b7a8fe261024fb33ee76fcd82d8cb4ba1d7870c2
* Result: FAILED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=41fb0887-ef27-4c00-ba10-bee2f44d7f94%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: b76652cc51a5716ffe54bf81a23510cc1d76d9f5
* Result: FAILED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=1fb79405-db3a-41c4-b56d-269c2537cff6%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: 01372c6f12b468ae7002654d327218c3525b036a
* Result: FAILED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=cf57796d-4cb9-485c-b7ba-85a4accd37a5%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

TheRealAmazonKendra commented:
Also, we recognize that some of the error messages emitted can be hard to understand. Please let us know if you need any clarification to get the build passing.

TheRealAmazonKendra commented:
@Mergifyio update

seanyles-amzn commented:
> Also, we recognize that some of the error messages emitted can be hard to understand. Please let us know if you need any clarification to get the build passing.

The linter check is simple enough to understand but the logs from CodeBuild are much too large for me to identify the issue. I could use some help on that front

TheRealAmazonKendra commented:
Looks like the issue is that the props in a few of the interfaces aren't `readonly`.

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: af0f72e4e15992356c2564c2621b152e1a82729b
* Result: FAILED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=993f6fa3-5947-4974-b77b-7eb3326ed10a%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

TheRealAmazonKendra reviewed (dismissed):
Looks like something with this merge went quite wrong. In the future, please don't use force push and rebase with the command in console `@Mergifyio update`.

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: 6b7ad6b605057972297e8879f0ff767545d1ce9a
* Result: SUCCEEDED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=1cec6518-fa7a-4ee9-9750-502316c26dba%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

seanyles-amzn commented:
I am git-illiterate aparently

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: 211cae8637ebbcee5c83cb758d61c989cd7b3409
* Result: FAILED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=03c59397-dfcf-433d-8e16-ae8f48532f76%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: 22c001cc5d057bb5d3847df1120d1812fe603460
* Result: FAILED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=08f49196-b40f-45f9-a33b-0d104930f88d%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: 6b7ad6b605057972297e8879f0ff767545d1ce9a
* Result: SUCCEEDED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=cd12964f-0956-4c4a-9d46-ac70275d64eb%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: a493cd460db1ff654e8a67a0579a45ba3d77d6db
* Result: FAILED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=bc0692bc-4431-48c4-8481-6905429686c2%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: 2aecb22d2b9e4ff0c8c95f7109cc3273388fda42
* Result: FAILED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=908dde61-fe32-4831-9bfa-bb673e03cdcd%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: de9b41f60914758905d3deec820a2f8f17f4bbd5
* Result: FAILED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=289da57e-b8d3-431c-be7a-673d88c87ff8%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: 62b22c0dfc2529ee1f7c888c831664d625477984
* Result: FAILED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=9ef178c9-d691-41be-8311-6f72ceffd63a%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

TheRealAmazonKendra commented:
@Mergifyio update

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: e5d68eecd06f06dc07a36cf3b27e90fed23791b2
* Result: FAILED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=186a6bdc-b811-4d4c-84ae-f6d0f1b373f0%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: 5a44f00be65e314eac34ca88f45534b7541b3105
* Result: FAILED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=1f0c777e-1fad-4cfe-add9-926cc50abf0f%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: df05c7e928b307b35267141648ace20ed27f25cc
* Result: FAILED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=8257e088-6006-4f49-b92b-ff1bdd9cde9e%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: 968eb96824edccce47a160b804fb5a3a64c25716
* Result: FAILED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=cde47744-b3ae-4a55-b93c-667ff73d68f0%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: 98763414d46734329b69e5fa14896ead5ae3ea49
* Result: FAILED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=dab39068-d1d0-4f10-a947-fdd93a8de331%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: 028c3baeae86ac831f3f5266f3cc3224b85f7b65
* Result: FAILED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=6d685e02-b2a5-41fe-bcc9-6b0a3ab00b83%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: 66da38be72e1480eb0c20728cd62b8c0cabee2d0
* Result: FAILED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=b3fb4210-56ef-4920-a5d3-dbe8cf63ba90%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: 0bc448ae8dd5e0b9d0a8265b6f328547d012c4be
* Result: FAILED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=d99f27f8-2103-4058-81b3-00ceddd43d41%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

seanyles-amzn commented:
> [awslint:no-unused-type:@aws-cdk/aws-cloudwatch.InsightRuleMetrics] type or enum is not used by exported classes (declared at lib/insight-rule.ts:11)

Not really sure what to do about this. This enum is used in the `metric` function of the InsightRule so that a user doesn't need to know the specific string metricName

seanyles-amzn commented:
https://github.com/Mergifyio update

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: 36f84bf9f13bc5bfa9cb453d76008e5f5184d7e3
* Result: FAILED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=b7eb1c96-5068-4e80-9143-b2efaa7adc3f%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

seanyles-amzn commented:
> > [awslint:no-unused-type:@aws-cdk/aws-cloudwatch.InsightRuleMetrics] type or enum is not used by exported classes (declared at lib/insight-rule.ts:11)
> 
> Not really sure what to do about this. This enum is used in the `metric` function of the InsightRule so that a user doesn't need to know the specific string metricName

@TheRealAmazonKendra

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: 266588a7f6019206ab4d321d1729a237615531ee
* Result: FAILED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=9a4958ea-e58f-47ad-839b-23bb1fc293ce%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

seanyles-amzn commented:
https://github.com/Mergifyio update

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: a158ec6e25cc9c14d7de7c357991d56797c7d6bf
* Result: FAILED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=179184d8-cfef-4448-a057-e3b7d7151416%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: 476f09bc9f95054d8606970e111ca085513c0862
* Result: FAILED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=65743544-9821-4ed5-8b44-852f909fa013%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: 426eaa4bb74af2e2c4d7b8208792fab8124d6848
* Result: FAILED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=ea538aab-b26f-4466-a093-1dc534fe5ab2%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: d6cb9d12dc58d13f06467431209af1d74a80e0cc
* Result: FAILED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=bfe8c194-94c2-44ac-b320-cb581b7ba23e%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

seanyles-amzn commented:
> @aws-cdk/aws-cloudwatch: error: [awslint:resource-attribute:@aws-cdk/aws-cloudwatch.InsightRule.insightRuleRuleName] resources must represent all cloudformation attributes as attribute properties. "@attribute ATTR[,ATTR]" can be used to tag non-standard attribute names. missing property: insightRuleRuleName

I don't understand this @attribute ATTR[,ATTR] to change the propName

@TheRealAmazonKendra

kaizencc commented on packages/@aws-cdk/aws-cloudwatch/lib/insight-rule.ts:
```diff
@@ -0,0 +1,366 @@
+   *
+   * @attribute InsightRuleRuleName
+   */
+  public readonly insightRuleName: string;
```
```suggestion
  public readonly insightRuleRuleName: string;
```

kaizencc reviewed (commented):
@seanyles-amzn have you tried this? See [design guidelines](https://github.com/aws/aws-cdk/blob/main/docs/DESIGN_GUIDELINES.md#attributes) doc for more on attributes in CDK.

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: ce59014ed009282fd63c7d55134508e1350a50c1
* Result: FAILED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=72a35ac4-1764-43dd-9471-1f6d03f5e274%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: 44232f0542b884d04528b96385bbe36a2ccfc81c
* Result: FAILED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=53eb4c1d-54f2-40ff-b9a8-f94be33cc67b%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: ea3fa6dc00197b0b64cb3a62988adb2b28ca8c99
* Result: FAILED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=812038c1-2702-4f04-8038-049712d64b72%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: a0f777687b9b3f1eefeb177f88191ec7f33a7330
* Result: FAILED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=2bccca1a-6677-4d73-9754-467e2ef86aa4%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

seanyles-amzn commented:
```
@aws-cdk/aws-cloudwatch:   CHANGED    integ.alarm-and-dashboard 1.544s
@aws-cdk/aws-cloudwatch:       Resources
@aws-cdk/aws-cloudwatch: [~] AWS::CloudWatch::Dashboard DashCCD7F836 
@aws-cdk/aws-cloudwatch:  â””â”€ [~] DashboardBody
@aws-cdk/aws-cloudwatch:      â””â”€ [~] .Fn::Join:
@aws-cdk/aws-cloudwatch:          â””â”€ @@ -83,6 +83,10 @@
@aws-cdk/aws-cloudwatch:             [ ]         "QueueName"
@aws-cdk/aws-cloudwatch:             [ ]       ]
@aws-cdk/aws-cloudwatch:             [ ]     },
@aws-cdk/aws-cloudwatch:             [-]     "\"]],\"singleValueFullPrecision\":true}},{\"type\":\"custom\",\"width\":6,\"height\":6,\"x\":0,\"y\":56,\"properties\":{\"endpoint\":\"arn:aws:lambda:us-west-2:123456789012:function:my-function\",\"title\":\"My custom alarm\",\"updateOn\":{\"refresh\":true,\"resize\":true,\"timeRange\":true}}}]}"
@aws-cdk/aws-cloudwatch:             [+]     "\"]],\"singleValueFullPrecision\":true}},{\"type\":\"custom\",\"width\":6,\"height\":6,\"x\":0,\"y\":56,\"properties\":{\"endpoint\":\"arn:aws:lambda:us-west-2:123456789012:function:my-function\",\"title\":\"My custom alarm\",\"updateOn\":{\"refresh\":true,\"resize\":true,\"timeRange\":true}}},{\"type\":\"metric\",\"width\":6,\"height\":6,\"x\":0,\"y\":62,\"properties\":{\"insightRule\":{\"maxContributorCount\":10,\"orderBy\":\"Sum\",\"ruleName\":\"my-insight-rule\"},\"legend\":{\"position\":\"bottom\"},\"period\":300,\"region\":\"",
@aws-cdk/aws-cloudwatch:             [+]     {
@aws-cdk/aws-cloudwatch:             [+]       "Ref": "AWS::Region"
@aws-cdk/aws-cloudwatch:             [+]     },
@aws-cdk/aws-cloudwatch:             [+]     "\",\"title\":\"My Insight Widget\",\"view\":\"timeSeries\"}}]}"
@aws-cdk/aws-cloudwatch:             [ ]   ]
@aws-cdk/aws-cloudwatch:             [ ] ]
@aws-cdk/aws-cloudwatch:   UNCHANGED  integ.composite-alarm 1.593s
@aws-cdk/aws-cloudwatch:   UNCHANGED  integ.alarm-with-label 1.616s
@aws-cdk/aws-cloudwatch: Snapshot Results: 
@aws-cdk/aws-cloudwatch: Tests:    1 failed, 5 total
@aws-cdk/aws-cloudwatch: Error: Some tests failed!
```
This appears to be a snapshot that is failing because I updated it? How do I run the build to accept my changes?  I get `MODULE_NOT_FOUND` when trying `yarn integ-runner --update-on-failed` even after running `yarn install`

aws-cdk-automation reviewed (changes requested):
The Pull Request Linter fails with the following errors:

	❌ Features must contain a change to an integration test file and the resulting snapshot.

PRs must pass status checks before we can provide a meaningful review.
